client/packages/inventory/src/Stocktake/ListView/ListView.tsx

```diff
@@ -1,92 +1,119 @@
-  useEffect(() => {
-    const disabledRows = rows?.filter(isStocktakeDisabled).map(({ id }) => id);
-    if (disabledRows) setDisabledRows(disabledRows);
-  }, [rows]);
-};
-
-export const StocktakeListView = () => {
+export const ListView = () => {
   const t = useTranslation();
   const navigate = useNavigate();
   const { localisedDate } = useFormatDateTime();
   const { user } = useAuthContext();
   const {
-    updateSortQuery,
-    updatePaginationQuery,
     filter,
-    queryParams: { sortBy, page, first, offset },
+    queryParams: { sortBy, first, offset },
   } = useUrlQueryParams({
     filters: [{ key: 'status', condition: 'equalTo' }],
   });
-  const pagination = { page, first, offset };
   const queryParams = { ...filter, sortBy, first, offset };
 
   const {
-    query: { data, isError, isLoading },
+    query: { data, isError, isFetching },
     hasStocktake,
   } = useStocktakeList(queryParams);
   const {
     create: { create, isCreating },
   } = useStocktake();
-  useDisableStocktakeRows(data?.nodes);
 
   const statusTranslator = getStocktakeTranslator(t);
   const description = t('stocktake.description-template', {
     username: user ? user.name : 'unknown user',
     date: localisedDate(new Date()),
   });
 
-  const columns = useColumns<StocktakeRowFragment>(
-    [
-      GenericColumnKey.Selection,
-      [
-        'stocktakeNumber',
-        { maxWidth: 75, sortable: false, defaultHideOnMobile: true },
-      ],
-      [
-        'status',
-        {
-          accessor: ({ rowData }) =>
-            rowData.isLocked
-              ? t('label.stocktake-on-hold')
-              : statusTranslator(rowData.status),
-        },
-      ],
-      ['description', { sortable: false }],
-      ['createdDatetime', { format: ColumnFormat.Date }],
-      ['stocktakeDate', { sortable: false, defaultHideOnMobile: true }],
-      getCommentPopoverColumn(),
+  const columns = useMemo(
+    (): ColumnDef<StocktakeRowFragment>[] => [
+      {
+        accessorKey: 'stocktakeNumber',
+        header: t('label.number'),
+        defaultHideOnMobile: true,
+        enableSorting: true,
+      },
+      {
+        id: 'status',
+        header: t('label.status'),
+        enableSorting: true,
+        accessorFn: row =>
+          row.isLocked
+            ? t('label.stocktake-on-hold')
+            : statusTranslator(row.status),
+        enableColumnFilter: true,
+        filterVariant: 'select',
+        filterSelectOptions: [
+          { label: t('status.new'), value: StocktakeNodeStatus.New },
+          {
+            label: t('status.finalised'),
+            value: StocktakeNodeStatus.Finalised,
+          },
+        ],
+      },
+      {
+        accessorKey: 'description',
+        header: t('label.description'),
+      },
+      {
+        accessorKey: 'createdDatetime',
+        header: t('label.created'),
+        columnType: ColumnType.Date,
+        enableSorting: true,
+      },
```

Comment:
Do we want to add a DateRange filter on this column?

Reply:
Yeah nice - mostly have just been focused on replacing existing filters, but this is probably a nice add!